### PR TITLE
fix(upgrade): silent bootstrap failures (#12062)

### PR DESCRIPTION
**Pull request by btrigueiro**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
#12062 (previous PR was #12232 but I messed up 😳)

**What is the new behavior?**
`bootstrap` now rejects the promise on errors and delegates the error handling to a common `onError` util (effectively logging the error and throwing it back)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by btrigueiro** [comment]
@vicb I've added tests for the thrown error and logged message but I've rolled back the `}, reject)` simplification as it indeed silenced the `throw`s (as noted by @andrei-ivanovici and @andreialecu). Let me know if anything else is needed.

File: modules/@angular/upgrade/test/upgrade_spec.ts [hunk 1]
```diff
@@ -20,6 +20,65 @@ export function main() {
 
     it('should have angular 1 loaded', () => expect(angular.version.major).toBe(1));
 
+
```
Comment (vicb) [inline modules/@angular/upgrade/test/upgrade_spec.ts]: nit: please remove the extra blank lines

File: modules/@angular/upgrade/test/upgrade_spec.ts [hunk 2]
```diff
@@ -20,6 +20,65 @@ export function main() {
...
+
+    describe('errors', () => {
+      let adapter: UpgradeAdapter;
+
+      beforeEach(() => {
+        const ng1Module = angular.module('ng1', []);
+
+        const Ng2Component =
```
Comment (vicb) [inline modules/@angular/upgrade/test/upgrade_spec.ts]: ng2Component (lowercase "n") - applicable at multiple places

File: modules/@angular/upgrade/test/upgrade_spec.ts [hunk 3]
```diff
@@ -20,6 +20,65 @@ export function main() {
...
+
+        const Ng2Module = NgModule({declarations: [Ng2Component], imports: [BrowserModule]}).Class({
+          constructor: function() {}
+        });
+
+        adapter = new UpgradeAdapter(Ng2Module);
+      });
+
+
+      it('should be thrown', fakeAsync(() => {
```
Comment (vicb) [inline modules/@angular/upgrade/test/upgrade_spec.ts]: - please find a better description than 'should be thrown', ie 'should throw when ...'

**Comment by vicb** [comment]
Thank you very much, this is super helpful.

Please fix the comments.

File: modules/@angular/upgrade/test/upgrade_spec.ts (hunk 1 above)
Comment (btrigueiro) [inline modules/@angular/upgrade/test/upgrade_spec.ts]: Cleaned up.. It's just that have this thing with extra organizing space 😁

File: modules/@angular/upgrade/test/upgrade_spec.ts (hunk 3 above)
Comment (btrigueiro) [inline modules/@angular/upgrade/test/upgrade_spec.ts]: Changed the test descriptions so it goes like
"adapter: ng1 to ng2 ... bootstrap errors ... should throw an uncaught error" and
"adapter: ng1 to ng2 ... bootstrap errors ... should properly log to the console and re throw"

I had to use `fakeAsync` to call `flushMicrotasks` and capture the `throw`.

File: modules/@angular/upgrade/test/upgrade_spec.ts [hunk 4]
```diff
@@ -20,6 +20,65 @@ export function main() {
...
+           expect(resolveSpy).not.toHaveBeenCalled();
+         }));
+
+
+      it('should be properly logged', fakeAsync(() => {
+           const element = html('<ng2></ng2>');
+
+           spyOn(console, 'log');
+
+           try {
```
Comment (btrigueiro) [inline modules/@angular/upgrade/test/upgrade_spec.ts]: Nice!

**Pull request closed without merging**